### PR TITLE
chore: add .gitpod.yml and .docker/gitpod/Dockerfile

### PR DESCRIPTION
**Pull request by austinletson**
Add docker files and .gitpod.yml to enable use of Gitpod with std4. Both of these are copied from mathlib with modifications for std4

I have opened std4 in Gitpod and tested,
- building std4
- running tests with make
- building and view docs

related to #565

### CODE REVIEW

**Comment by austinletson** [comment]
awaiting-review

File: .docker/gitpod/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This is the Dockerfile for leanprover/std4
+# This file is mostly copied from [mathlib4](https://github.com/leanprover-community/mathlib4/blob/master/.docker/gitpod/Dockerfile)
+
+# gitpod doesn't support multiple FROM statements, (or rather, you can't copy from one to another)
+# so we just install everything in one go
+FROM ubuntu:jammy
+
+USER root
+
+RUN apt-get update && apt-get install sudo git curl git bash-completion python3 python3-requests gcc make -y && apt-get clean
```
Comment (semorrison) [inline .docker/gitpod/Dockerfile:10]: I don't think we should need any python3 anymore. (Ideally we'd also remove this on the mathlib side).

There's also a repeated `git` here.

Could you test without?

Comment (austinletson) [reply .docker/gitpod/Dockerfile]: I have removed `git` and `python3` and ran everything again successfully. 

I didn't remove `python3-requests`. We still need that right? If you remove it you can't run `python3 -m http.server` to expose the documentation.

**Comment by austinletson** [comment]
awaiting-review

File: .docker/gitpod/Dockerfile (hunk 1 above)
Comment (semorrison) [reply .docker/gitpod/Dockerfile]: I guess --- I've never done that. I guess it doesn't hurt too much having python bundled in.

Comment (austinletson) [reply .docker/gitpod/Dockerfile]: Alright, my thinking was that all the commands in the README.md should work out of the box.

I just launched Gitpod successfully from https://github.com/leanprover/std4 and built std4. Thanks!

**Pull request merged**